### DISCUSSION TITLE
SSP filtering for gap analysis

### DISCUSSION THREAD

**Discussion by jpower432** [oscal-compass/compliance-trestle#1331] [Ideas] [answered]
Hi `compliance-trestle` maintainers,

I'd like to use the 'trestle author ssp-filter' to find controls in the profile associated with the SSP that don't have a control implementation defined. I see that the current usage supports filtering by a specific subset of controls or components. Would filtering by control implementation status be appropriate for this sub-command's intended use? If not, is there a recommended approach using existing tool functionality? I am currently testing a possible solution using the `jinja` command.

Thanks in advance!

cc @fsuits

**Comment by fsuits** [comment, marked as answer]
Hi @jpower432 !  There isn't currently a way to filter based on status, but that does make sense as a new feature.  I will create an issue for it.  Thanks!

I don't know a convenient way to do what you request right now, but if some python writing is allowed, it wouldn't be too hard to load an ssp json file and go through it to remove based on status - then save it out.  Let me know if you are interested in that - and I will make an issue either way.

**Reply by jpower432** [reply]
Hi @fsuits. Thanks for the reply! I can get by for now using the `compliance-trestle` libraries. I have been using them in some Python scripts for testing custom logic.
